pytools/graph.py

```diff
@@ -199,26 +203,51 @@ class HeapEntry:
     Only needs to define :func:`pytools.graph.__lt__` according to
     <https://github.com/python/cpython/blob/8d21aa21f2cbc6d50aab3f420bb23be1d081dac4/Lib/heapq.py#L135-L138>.
     """
-    def __init__(self, node, key):
-        self.node = node
-        self.key = key
+    node: Any
+    key: Any
```

Comment:
Maybe introduce type variables for key and node, for increased expressiveness of the type annotation?